### PR TITLE
Fix Full scan to be using the python driver

### PR DESCRIPTION
**Pull request by juliayakovlev**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [ ] ~~All new and existing unit tests passed (CI)~~
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

Run full table scan using python driver with paging on. 
Run with CL=ONE. Every new scan will run from different node (randomly) .

I didn't run full test as we have troubles with spots.
I ran the test for a few hours and see that one node was worked harder then others - because of full scan

### CODE REVIEW

**Comment by roydahan** [comment]
Does it woron on longevity 50GB? 200GB? 1TB?

**Comment by juliayakovlev** [comment]
> Does it woron on longevity 50GB? 200GB? 1TB?

As I wrote in description:
I didn't run full test as we have troubles with spots.
I ran the test for a few hours and see that one node was worked harder then others - because of full scan

File: sdcm/tester.py [hunk 1]
```diff
@@ -891,8 +893,42 @@ def verify_gemini_results(self, queue):
...
+
+        FullScanEvent(type='start', db_node_ip=db_node.ip_address, ks_cf=ks_cf)
+
+        try:
+            with self.cql_connection_patient(db_node) as session:
+                result = session.execute(SimpleStatement('select * from {}'.format(ks_cf), fetch_size=page_size,
+                                                         consistency_level=ConsistencyLevel.ONE))
+                while result.has_more_pages:
+                    result.fetch_next_page()
+            result = {'exit_code': 0, 'stdout': 'finished successfully', 'stderr': ''}
```
Comment (bentsi) [inline sdcm/tester.py]: no need to maintain the same structure as it ran in a subprocess, adapt this to the new code

File: sdcm/tester.py [hunk 2]
```diff
@@ -905,10 +941,30 @@ def run_fullscan_thread(self, ks_cf='random', interval=1, duration=None):
...
-                                         interval=interval,
-                                         duration=duration)
+
+        @log_run_info('Fullscan thread')
+        def run_in_thread():
+            start = current = time.time()
+            while current - start < duration:
+                loader_node = random.choice(self.loaders.nodes)
+                db_node = random.choice(self.db_cluster.nodes)
+                if db_node.termination_event.isSet():
```
Comment (bentsi) [inline sdcm/tester.py]: we can improve and move this in the while condition check

File: sdcm/tester.py [hunk 3]
```diff
@@ -905,10 +941,30 @@ def run_fullscan_thread(self, ks_cf='random', interval=1, duration=None):
...
+
+                self.run_fullscan(ks_cf, loader_node, db_node)
+                time.sleep(interval)
+                current = time.time()
+
+        th = threading.Thread(target=run_in_thread)
+        th.daemon = True
+        th.start()
+
+    def kill_fullscan_thread(self):
```
Comment (bentsi) [inline sdcm/tester.py]: don't think we need this, `termination_event` is general for test end, remove this and a call  this method

File: sdcm/tester.py (hunk 2 above)
Comment (juliayakovlev) [reply sdcm/tester.py]: @bentsi  could you explain please?

Comment (juliayakovlev) [reply sdcm/tester.py]: I want that every time to run the full scan from another node, not from same

File: sdcm/tester.py (hunk 1 above)
Comment (juliayakovlev) [reply sdcm/tester.py]: Fixed

File: sdcm/tester.py (hunk 3 above)
Comment (juliayakovlev) [reply sdcm/tester.py]: Fixed

**Comment by juliayakovlev** [comment]
@bentsi  please, review

File: sdcm/tester.py (hunk 2 above)
Comment (bentsi) [reply sdcm/tester.py]: ok, looking on it again, this is an issue, Consider
We don't need to run on db_node when terminate and replace/decommission nemesis run 
and need ability to terminate the thread at the end of the test

File: sdcm/tester.py [hunk 4]
```diff
@@ -891,8 +892,41 @@ def verify_gemini_results(self, queue):
...
+        try:
+            with self.cql_connection_patient(db_node) as session:
+                result = session.execute(SimpleStatement('select * from {}'.format(ks_cf), fetch_size=page_size,
+                                                         consistency_level=ConsistencyLevel.ONE))
+                while result.has_more_pages:
+                    result.fetch_next_page()
+            FullScanEvent(type='finish', db_node_ip=db_node.ip_address, ks_cf=ks_cf, message='finished successfully')
+        except Exception as details:
+            FullScanEvent(type='finish', db_node_ip=db_node.ip_address, ks_cf=ks_cf, message=details.message,
+                          severity=Severity.CRITICAL)
```
Comment (bentsi) [inline sdcm/tester.py]: per our discussion make the severity warning

Comment (juliayakovlev) [reply sdcm/tester.py]: Fixed

**Comment by juliayakovlev** [comment]
@bentsi  Please, review

**Pull request merged**